src/PreviewGroup.tsx

```diff
@@ -42,17 +42,25 @@ const Group: React.FC<GroupConsumerProps> = ({
   const [mousePosition, setMousePosition] = useState<null | { x: number; y: number }>(null);
 
   const registerImage = (id: number, url: string) => {
-    setPreviewUrls(oldPreviewUrls => {
-      return new Map(oldPreviewUrls).set(id, url);
-    });
-
-    return () => {
+    const unRegister = () => {
       setPreviewUrls(oldPreviewUrls => {
         const clonePreviewUrls = new Map(oldPreviewUrls);
         const deleteResult = clonePreviewUrls.delete(id);
         return deleteResult ? clonePreviewUrls : oldPreviewUrls;
       });
     };
+
+    // we don't to test this if canPreview changed when same url
```

Comment:
```suggestion
    // we don't need to test this if canPreview changed when url stays the same
```
Maybe this is better👀

Reply:
Thanks.